File: Backend/src/services/scoring.ts
export type Answer = {
  questionId: string
  value: number
  category: "emotional" | "cognitive" | "physical" | "support"
  reverse?: boolean
}

export function calculateBurnoutScore(answers: Answer[]) {
  let total = 0
  const categoryTotals: Record<string, number> = {}

  for (const a of answers) {
    const value = a.reverse ? 6 - a.value : a.value
    total += value
    categoryTotals[a.category] =
      (categoryTotals[a.category] || 0) + value
  }

  const score = Math.round((total / 60) * 100)

  const severity =
    score >= 67 ? "high" :
    score >= 34 ? "medium" :
    "low"

  return {
    score,
    severity,
    categoryTotals
  }
}

File: Backend/app/services/scoring.py
def calculate_burnout_score(answers: dict) -> dict:
    """Calculate burnout score from assessment answers"""
    reverse_scored = ["q10", "q11"]

    total = 0
    max_total = 0

    for q, value in answers.items():
        max_total += 5
        if q in reverse_scored:
            value = 6 - value
        total += value

    score = int((total / max_total) * 100)

    if score < 33:
        severity = "low"
        summary = "Low indicators of burnout."
    elif score < 66:
        severity = "moderate"
        summary = "Moderate burnout indicators."
    else:
        severity = "high"
        summary = "High burnout indicators."

    return {
        "score": score,
        "severity": severity,
        "summary": summary
    }